pootle/core/browser.py

```diff
@@ -100,7 +100,12 @@ def make_generic_item(path_obj):
 
 
 def make_directory_item(directory):
-    item = make_generic_item(directory)
+    filters = {}
+
+    if len(VirtualFolder.get_matching_for(directory.pootle_path)):
```

Comment:
This has been replaced by a new call that triggers only 1 query.

Regarding the cache, the plan is to get something that works, and then optimize it using cache.